### PR TITLE
fix: use `modulePaths` to improve Jest 27 imports

### LINKED ISSUE #219: bug: ESLint cannot be imported in Jest 27
- `kcd-scripts` version: 11.2.0
- `node` version: 16.9.1
- `npm` (or `yarn`) version: 7.23.0
- `jest` version: 27.2.0

Relevant code or config

```javascript
import 'eslint'
// or
require('eslint')
```

What you did: `kcd-scripts test`

What happened:

```
[test]   ● Test suite failed to run
[test] 
[test]     TypeError: Ajv is not a constructor
[test] 
[test]       at Object.<anonymous>.module.exports (node_modules/@eslint/eslintrc/lib/shared/ajv.js:19:17)
```

Reproduction repository:

https://github.com/testing-library/eslint-plugin-jest-dom/tree/5cfb8c97ecc78fb4f61748fe083123bed7dc1f9e

Alternatively to reproduce without `kcd-scripts`, use the following Jest config:

```js
// jest.config.js
module.exports = {
  moduleDirectories: ['node_modules', 'shared'],
}
```

Problem description: `kcd-scripts` adds `'shared'` to `moduleDirectories` (I assume some projects might rely on this for absolute imports. Unfortunately, there is a file in ESLint in a directory called `shared` which has the same name as a package, and when it tries to import that package, the config causes it to import itself, and ESLint breaks with a runtime error because the script does not have the same API as the package. This blocked us from upgrading to Jest 27 in `eslint-plugin-jest-dom` (https://github.com/testing-library/eslint-plugin-jest-dom/pull/185), which has tests relying on the ESLint rule tester.

Suggested solution: It seems like the most straightforward solution is to remove `'shared'` from `'moduleDirectories'`. However I'm concerned this could break back compatibility with projects that use a `shared` directory internally. Ideally we should figure out a way to configure Jest's module resolution so both ESLint imports and `shared` imports work, but I don't know how to do that myself, and reordering `moduleDirectories` doesn't seem to help either.

### PR DESCRIPTION
**Pull request by nickmccurdy**
**What**: Use `modulePaths` to improve Jest 27 imports

**Why**: Fixes #219 which would unblock https://github.com/testing-library/eslint-plugin-jest-dom/pull/185

**How**: 
- Moved all `moduleDirectories` (except `node_modules`) to `modulePaths`, which properly prioritizes resolving `node_modules` over local project modules
  - It's theoretically possible that this could break a test that relies on the opposite functionality, though IMO that would not be a good assumption to make, and it's worth noting that semver does not protect users from breaking changes resulting from bug fixes
- Switched to using `<rootDir>`, which should provide better support for `kcd-scripts test` in monorepos
- Ran an end to end test using `npm link` within [testing-library/eslint-plugin-jest-dom](https://github.com/testing-library/eslint-plugin-jest-dom)
  - The entire test suite now passes with Jest 26 (previously passing) and 27 (previously failing)

**Checklist**:

- [ ] Documentation N/A
- [x] Tests
- [x] Ready to be merged

### CODE REVIEW

**Review by MichaelDeBoey — APPROVED** [review]

**Review by kentcdodds — APPROVED** [review]
Cool 👍 Thanks!

File: src/config/jest.config.js [hunk 1]
```diff
@@ -59,7 +54,7 @@ const setupFiles = [
 ]
 for (const setupFile of setupFiles) {
   if (hasFile(setupFile)) {
-    jestConfig.setupFilesAfterEnv = [fromRoot(setupFile)]
+    jestConfig.setupFilesAfterEnv = `<rootDir>/${setupFile}`
```
Comment (nstepien) [inline src/config/jest.config.js:57]: https://github.com/testing-library/dom-testing-library/runs/3611964206
```
[test] ● Validation Error:
[test] 
[test]   Option "setupFilesAfterEnv" must be of type:
[test]     array
[test]   but instead received:
[test]     string
[test] 
[test]   Example:
[test]   {
[test]     "setupFilesAfterEnv": [
[test]       "<rootDir>/testSetupFile.js"
[test]     ]
[test]   }
[test] 
[test]   Configuration Documentation:
[test]   https://jestjs.io/docs/configuration
```

Comment (nstepien) [reply src/config/jest.config.js:57]: Fixed in #221

Comment (nickmccurdy) [reply src/config/jest.config.js:57]: Oops I forgot to test that use case, thanks for the fix!

**Pull request merged**